modules/fs2/src/test/scala/pureconfig/module/fs2Suite.scala

```diff
@@ -26,7 +26,7 @@ class fs2Suite extends AnyFlatSpec with Matchers {
   implicit val reader: ConfigReader[SomeCaseClass] =
     ConfigReader.forProduct2("somefield", "anotherfield")(SomeCaseClass.apply)
   implicit val writer: ConfigWriter[SomeCaseClass] =
-    ConfigWriter.forProduct2("somefield", "anotherfield")(SomeCaseClass.unapply(_).get)
+    ConfigWriter.forProduct2("somefield", "anotherfield")(x => (x.somefield, x.anotherfield))
```

Comment:
Out of curiosity, why did you have to change this to be an explicit application? Is there a way to keep using `unapply`?

Reply:
You can keep using `unapply` in both Scala 2 and Scala 3, but the code will not cross-compile across Scalas. This is because in Scala 2 it returns an `Option` but in Scala 3 it is "option-less" .

https://docs.scala-lang.org/scala3/reference/changed-features/pattern-matching.html